test/process/index/IndexEntryStorageStrategyTest.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "common/CommonStructsAndConstants.h"
+#include "process/index/IndexEntryStorageStrategy.h"
+#include <UnitTest++/UnitTest++.h>
+#include <iostream>
+
+const char *const INDEXENTRY_STORAGESTRATEGY_SUITE_TESTS = "IndexEntryStorageStrategyTests";
+const char *const TEST_BLOCKSTRATEGY_CALCULATE_BLOCK_INTERVAL = "Test calculateBlockInterval";
+const char *const TEST_BLOCKSTRATEGY_SHALLSTORE = "Test calculateBlockInterval";
+const char *const TEST_BYTESTRATEGY_PARSESTRINGVALUE = "Test parseStringValue";
+const char *const TEST_BYTESTRATEGY_CONSTRUCT = "Test construction";
+const char *const TEST_BYTESTRATEGY_SHALLSTORE = "Test shallStore with a valid byte distances";
+const char *const TEST_BYTESTRATEGY_SHALLSTORE_INVALID_BYTEDISTANCE = "Test shallStore with a byte distance of -1";
+
+SUITE (INDEXENTRY_STORAGESTRATEGY_SUITE_TESTS) {
+
+    TEST (TEST_BLOCKSTRATEGY_CALCULATE_BLOCK_INTERVAL) {
+                CHECK_EQUAL(16U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(1 * GB));
+                CHECK_EQUAL(32U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(2 * GB));
+                CHECK_EQUAL(64U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(4 * GB));
+                CHECK_EQUAL(128U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(8 * GB));
+                CHECK_EQUAL(512U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(20 * GB));
+                CHECK_EQUAL(512U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(30 * GB));
+                CHECK_EQUAL(1024U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(40 * GB));
+                CHECK_EQUAL(2048U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(80 * GB));
+                CHECK_EQUAL(2048U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(120 * GB));
+                CHECK_EQUAL(4096U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(160 * GB));
+                CHECK_EQUAL(4096U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(200 * GB));
+                CHECK_EQUAL(8192U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(300 * GB));
+        //Maximum value
+                CHECK_EQUAL(8192U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(430 * GB));
+                CHECK_EQUAL(8192U, BlockDistanceStorageStrategy::calculateIndexBlockInterval(1630 * GB));
+    }
+
+    TEST (TEST_BLOCKSTRATEGY_SHALLSTORE) {
+        BlockDistanceStorageStrategy strat(16, true);
+        auto ie0 = IndexEntryV1::from(0, 0, 16, 32, 0);
+        auto ie1 = IndexEntryV1::from(0, 15, 19, 512 * MB, 3);
```

Comment:
BTW: One used paradigm I have seen before is instead of constants and additionally including the multipliciation in the abstraction while yielding a declarative (unfortunately prefix) syntax, is:

```c++
inline long mb(long amount) {
   return kb(1024 * amount);
}

inline long kb(long amount) {
   return amount * 1024;
}

...
```